### ISSUE TITLE
Can't set CompletionMenu colors from highlighting_style_overrides

### ISSUE THREAD

**Issue by petobens** [ipython/ipython#11526]
Hi! I'm using [this ipython_config.py (https://github.com/petobens/dotfiles/blob/master/python/ipython_config.py) but I cannot seem to override the completion menu colors. Following https://github.com/petobens/dotfiles/blob/master/python/ipython_config.py I tried doing the following (see my config file for details):
```python
    Token.CompletionMenu: f'bg:{pmenu} {syntax_fg}',
    Token.CompletionMenu.Completion: f'bg:{pmenu} {syntax_fg}',
    Token.CompletionMenu.Completion.Current: f'bg:{light_blue} {black}',
    Token.CompletionMenu.Meta.Completion: f'bg:{pmenu} {syntax_fg}',
    Token.CompletionMenu.Meta.Completion.Current: f'bg:{light_blue} {black}',
    Token.CompletionMenu.MultiColumnMeta: f'bg:{light_blue} {black}',
```
but colors are not changed (whereas all the other colors are correctly overriden). I also tried using just `Token.Menu.Completions` but without success. I seem to recall that this used to work. I'm on `7.2.0`.

Thanks in advance for looking into this.

**Comment by petobens** [comment]
Applying the following patch:
```patch
--- pygments.py 2018-12-12 21:52:06.765154645 -0300
+++ pygments_mod.py      2018-12-12 21:54:27.775161086 -0300
@@ -45,7 +45,10 @@
     pygments_style = []

     for token, style in pygments_dict.items():
-        pygments_style.append((pygments_token_to_classname(token), style))
+        if isinstance(token, str):
+            pygments_style.append((token, style))
+        else:
+            pygments_style.append((pygments_token_to_classname(token), style))

     return Style(pygments_style)
```
and using prompt tookit syles as per https://github.com/prompt-toolkit/python-prompt-toolkit/blob/master/prompt_toolkit/styles/defaults.py#L74 fixes it for me. 

I'm pretty sure there is a better way of doing this but I can create a PR if necessary. @jonathanslenders

**Comment by memeplex** [comment]
Yes, this got broken recently. @petobens you should suggest your fix or start a discussion in https://github.com/prompt-toolkit/python-prompt-toolkit/issues instead, because this is not really an issue in ipython but upstream. Notice that, meanwhile, you can just monkey-patch the function if you feel like that (I don't, thus the following is untested):

```python
def my_style_from_pygments_dict(pygments_dict):
    pygments_style = []
    for token, style in pygments_dict.items():
        if isinstance(token, str):
            pygments_style.append((token, style))
        else:
            pygments_style.append((pygments_token_to_classname(token), style))
    return Style(pygments_style)
pygments.style_from_pygments_dict = my_style_from_pygments_dict
```

**Comment by jonathanslenders** [comment]
Hi @memeplex, @petobens,

If the following patch solves the issue, I would merge it into prompt_toolkit if you create a PR. It doesn't break anything, and it does solve the issue. (Unless there are some objections.)

It would require some documentation, because it's not really a Pygments dictionary anymore, but more a mixed pygments/prompt_toolkit style dictionary. We would also need both a PR against prompt_toolkit 2.0 branch as well as the master branch (prompt_toolkit 3.0).
